presto-hive/src/main/java/io/prestosql/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -153,7 +153,7 @@ public TableStatistics getTableStatistics(
             partitionsSample = getPartitionsSample(partitions, sampleSize);
             samplePartitionCache.put(table, new SamplePartition(partitions.size(), partitionsSample));
         }
-        else if (sample != null) {
```

Comment:
If  no else case then assert as assumption shouldnt remove the sanity check alone